omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -647,6 +648,14 @@ def gen_redundant(node_name, topology)
       end
     end
 
+    def ensure_stats_password
```

Comment:
The RFC for this feature says:

> A username and password will be generated if it not provided.

So, I think we should probably be treating this like the other non-optional passwords where we generate the username and password if they are not provided.

Reply:
Ah I missed that.  Ok - I need to figure out how to save a generated password.

Reply:
:+1: